src/parser/Parser.ts

```diff
@@ -1880,39 +1739,55 @@ export class Parser {
      *                    ignored.
      */
     private block(...terminators: BlockTerminator[]): Block | undefined {
+        this.consumeStatementSeparators(true);
         let startingToken = this.peek();
 
         const statements: Statement[] = [];
-        while (!this.checkAny(...terminators) && !this.isAtEnd()) {
+        while (!this.isAtEnd() && !this.checkAny(TokenKind.EndSub, TokenKind.EndFunction, ...terminators)) {
```

Comment:
Why are the `EndSub` and `EndFunction` terminators here? shouldn't those just be passed in from whatever is consuming the block? Will this cause issues when something does not want the block to stop at `EndSub` or `EndFunction`?

Reply:
The goal is to handle unterminated blocks when reaching the end of a function.